pymc3/model.py

```diff
@@ -1518,3 +1472,343 @@ def all_continuous(vars):
+                '`accept_cons_shared=True` to `add`.'.
+                format(type(node)))
+    if not isinstance(node, Hashable):
+        node = WrapAsHashable(node)
+    if node in dag:
+        # Should we raise a warning with we attempt to add a node that is
+        # already in the DAG??
+        if return_added_node:
+            return dag, node
+        else:
+            return dag
+
+    # Add node into the nodes set and then initiate all node relations and
+    # values to their defaults
+    dag.add_node(node)
+
+    # Try to get the conditional parents of node and add them
+    cond = get_first_level_conditionals(node)
+    if cond is not None:
+        for conditional_parent in walk_down_ownership(cond):
+            if conditional_parent not in dag:
+                try:
+                    add_to_dependence_dag(dag, conditional_parent)
+                except ConstantNodeException:
+                    continue
+            dag.add_edge(conditional_parent,
+                         node,
+                         conditional=True,
+                         deterministic=False)
+
+    # Try to get the deterministic parents of node and add them
+    if not_shared_or_constant_variable(node):
+        for deterministic_parent in walk_down_ownership([node], ignore=True):
+            if deterministic_parent not in dag:
+                try:
+                    add_to_dependence_dag(dag, deterministic_parent)
+                except ConstantNodeException:
+                    continue
+            dag.add_edge(deterministic_parent,
+                         node,
+                         conditional=False,
+                         deterministic=True)
+    if not networkx.is_directed_acyclic_graph(dag):
+        raise RuntimeError('The dependence graph is no longer a directed '
+                           'acyclic graph (DAG). The addition of node `{}`, '
+                           'of type `{}`, and the edges from its predecessors '
+                           'introduced a loop inside the variable dependence '
+                           'graph. Consider raising an issue with developers.'.
+                           format(node, type(node)))
+
+    if return_added_node:
+        return dag, node
+    return dag
+
+
+def walk_down_ownership(node_list, ignore=False):
+    """This function goes through an iterable of nodes provided in
+    `node_list`, yielding the non None named nodes in the ownership graph.
+    With the optional input `ignore`, a node without a name can be yielded.
+    """
+    for node in node_list:
+        if hasattr(node, 'name') and node.name is not None and not ignore:
+            yield node
+        elif not_shared_or_constant_variable(node):
+            owner = getattr(node, 'owner', None)
+            if owner is not None:
+                for parent in walk_down_ownership(owner.inputs):
+                    yield parent
+
+
+def get_sub_dag(dag, input_nodes, force=True, return_index=False):
```

Comment:
Here as well I think you always want to return the index for the way you use that function elsewhere.